scico/numpy/util.py

```diff
@@ -52,15 +50,6 @@ def ensure_on_device(
                 stacklevel=2,
             )
 
-        elif not isinstance(
```

Comment:
This might be controversial. My logic: instead of listing the types we are okay with, implicitly be okay with any type that can have `device_put` called on it.